include/oneapi/dpl/internal/random_impl/bernoulli_distribution.h

```diff
@@ -0,0 +1,193 @@
+    operator()(_Engine& __engine, const param_type& __params, unsigned int __random_nums)
+    {
+        return result_portion_internal<size_of_type_, _Engine>(__engine, __params, __random_nums);
+    }
+
+  private:
+    // Size of type
+    static constexpr int size_of_type_ = internal::type_traits_t<result_type>::num_elems;
+
+    // Static asserts
+    static_assert(::std::is_same<scalar_type, bool>::value,
+                  "oneapi::dpl::bernoulli_distribution. Error: unsupported data type");
+
+    // Distribution parameters
+    double p_;
+
+    // Implementation for generate function
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr != 0), result_type>::type
+    generate(_Engine& __engine, const param_type& __params)
+    {
+        return generate_vec<_Ndistr, _Engine>(__engine, __params);
+    }
+
+    // Specialization of the scalar generation
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr == 0), result_type>::type
+    generate(_Engine& __engine, const param_type& __params)
+    {
+        oneapi::dpl::uniform_real_distribution<double> __distr;
+        return __distr(__engine) < __params.p;
+    }
+
+    // Specialization of the vector generation with size = [1; 3]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N <= 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type& __params)
+    {
+        return generate_n_elems<_Engine>(__engine, __params, __N);
+    }
+
+    // Specialization of the vector generation with size = [4; 8; 16]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N > 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type& __params)
+    {
+        oneapi::dpl::uniform_real_distribution<sycl::vec<double, __N>> __distr;
+        sycl::vec<double, __N> __u = __distr(__engine);
+        sycl::vec<long, __N> __res_long = __u < sycl::vec<double, __N>{__params.p};
+        result_type __res;
+        for (int i = 0; i < __N; i++)
+            __res[i] = static_cast<bool>(__res_long[i]);
+        return __res;
+    }
+
+    // Implementation for the N vector's elements generation
+    template <class _Engine>
+    result_type
+    generate_n_elems(_Engine& __engine, const param_type& __params, unsigned int __N)
+    {
+        result_type __res;
+        oneapi::dpl::uniform_real_distribution<double> __distr;
+        for (int i = 0; i < __N; i++)
+            __res[i] = __distr(__engine) < __params.p;
+        return __res;
+    }
+
+    // Implementation for result_portion function
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr != 0), result_type>::type
+    result_portion_internal(_Engine& __engine, const param_type& __params, unsigned int __N)
+    {
+        result_type __part_vec;
+        if (__N == 0)
+            return __part_vec;
+        else if (__N >= _Ndistr)
+            return operator()(__engine);

```

Comment:
I think here should be called operator()(__engine, __params) method instead of operator()(__engine). 
Could you please check other distributions in case if there are also such issues.

Reply:
Yes, there is the same problem in all previous distributions including normal and uniform.

Reply:
done